pkg/kubelet/dockertools/docker.go

```diff
@@ -150,6 +150,31 @@ func filterHTTPError(err error, image string) error {
 	}
 }
 
+// Check if the inspected image matches what we are looking for
+func matchImageTagOrSHA(inspected dockertypes.ImageInspect, image string) bool {
+	parts := strings.SplitN(image, ":", 2)
+	if len(parts) == 1 {
+		// No Tag or SHA specified, so just return what we have
+		return true
+	}
+	// Check the RepoTags for an exact match
+	for _, tag := range inspected.RepoTags {
+		if tag == image {
+			// We found a specific tag that we were looking for
+			return true
+		}
+	}
+	// Look specifically for short and long sha(s)
+	if size := len(parts[1]); size == 12 || size == 64 {
```

Comment:
> If they give any other string we need to make sure that string is a tag and would not match a sha

My problem with that is if we don't validate the "Image" string in the apiserver, users can specify a string "sha256:.." of any length. In that case, should we still verify that the partial string matches the complete sha? I think we should do that anyway.